js/privatebin.js

```diff
@@ -1765,8 +1766,9 @@ jQuery.PrivateBin = function($, sjcl, Base64, RawDeflate) {
             }
 
             // set text
-            Helper.setElementText($plainText, text);
-            Helper.setElementText($prettyPrint, text);
+            var sanitizedText = DOMPurify.sanitize(text, {SAFE_FOR_JQUERY: true})
```

Comment:
Hmm, so you sanitize the text *before* actually passing. (for `$plainText` this is okay, as we never touch it later)
But especially using htmlEntities/urls2links later can break all sanitization as it converts some characters/stuff back to HTML.
Note that DOMPurify allows some links to pass through, so rather do htmlEntities/urls2links first, then sanitize. It should not break the links conversion or so.

I'd say if we apply it, always apply it as the last step. Then you are sure it is always sanitized.

Reply:
Sure, go ahead.